packages/base/src/tools.ts

```diff
@@ -452,3 +456,271 @@ export const loadGeoTIFFWithCache = async (sourceInfo: {
     sourceUrl: sourceInfo.url
   };
 };
+
+/**
+ * Generalized file reader for different source types.
+ *
+ * @param fileInfo - Object containing the file path and source type.
+ * @returns A promise that resolves to the file content.
+ */
+export const loadFile = async (fileInfo: {
+  filepath: string;
+  type: IJGISSource['type'];
+  model: IJupyterGISModel;
+}) => {
+  const { filepath, type, model } = fileInfo;
+
+  if (filepath.startsWith('http://') || filepath.startsWith('https://')) {
+    switch (type) {
+      case 'ImageSource': {
+        return filepath; // Return the URL directly
+      }
+
+      case 'ShapefileSource': {
+        try {
+          const response = await fetch(
+            `/jupytergis_core/proxy?url=${filepath}`
+          );
+          const arrayBuffer = await response.arrayBuffer();
+          const geojson = await shp(arrayBuffer);
+          return geojson;
+        } catch (error) {
+          console.error('Error loading remote shapefile:', error);
+          throw error;
+        }
+      }
+
```

Comment:
We should be able to implement the case for GeoJSON source as well? Simply going through the proxy similar as for shapefiles.

```typescript
case 'GeoJSONSource': {

```

Reply:
Done!